python/ql/lib/semmle/python/frameworks/Pyramid.qll

```diff
@@ -0,0 +1,87 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `pyramid` PyPI package.
+ * See https://docs.pylonsproject.org/projects/pyramid/.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+private import semmle.python.dataflow.new.FlowSummary
+private import semmle.python.frameworks.internal.PoorMansFunctionResolution
+private import semmle.python.frameworks.data.ModelsAsData
+
+/**
+ * Provides models for the `pyramid` PyPI package.
+ * See https://docs.pylonsproject.org/projects/pyramid/.
+ */
+module Pyramid {
+  // TODO: qldoc
+  module View {
+    /**
+     * A callable that could be used as a pyramid view callable.
+     */
+    private class PotentialViewCallable extends Function {
+      PotentialViewCallable() {
+        this.getPositionalParameterCount() = 1 and
+        this.getArgName(0) = "request"
```

Comment:
is there a hard requirement that the request parameter MUST be called `request`? or could it be named `req` as well?

Reply:
[Documentation](https://docs.pylonsproject.org/projects/pyramid/en/2.0-branch/narr/views.html) suggests that it should:
> View callables must, at a minimum, accept a single argument named request. 
however empirical testing seems to show it can allow other names.